ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogEntryIdAllocator.scala

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.validator
+
+import java.util.UUID
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.DamlLogEntryId
+import com.google.protobuf.ByteString
+
+trait LogEntryIdAllocator {
+  def allocate(): DamlLogEntryId
+}
+
+object LogEntryIdAllocator {
+  object Random extends LogEntryIdAllocator {
+    override def allocate(): DamlLogEntryId =
+      DamlLogEntryId.newBuilder
+        .setEntryId(ByteString.copyFromUtf8(UUID.randomUUID().toString))
```

Comment:
You may want to use the thread local variant to reduce contention.

Reply:
As it's only used in tests, I'm going to not bother. :upside_down_face:

Reply:
Alright, I propose to add a comment to let readers know.